map.js

```diff
@@ -4,35 +4,157 @@
 var olMap;
 
 /**
- * Places the point on the map
+ * Coordinate object
  *
- * @param ol.Map
- * @param {lat:, lng:} coords
+ * Normalizes data
+ * Provides transformation(s)
+ *
+ * @param {ol.MapEvent.coordinate|Number} - ol.MapEvent.coordinate or latitude
+ * @param {Number=} longitude
  */
-function placePoint(coords) {
-  var layer = olMap.getLayers().getArray()[1];
-  var point = new ol.Feature({
-    geometry: new ol.geom.Point(
-      ol.proj.fromLonLat([coords.lng, coords.lat])
-    )
+function Coordinate() {
+  var lonLat;
+
+  if (arguments.length < 2) {
+    lonLat = this._initCoordinate(arguments[0]);
+  } else {
+    lonLat = this._initLatLng(arguments[0], arguments[1]);
+  }
```

Comment:
To steal a previous idea from you, this could be simplified to ```lonLat = (arguments.length < 2) ? this._initCoordinate(arguments[0]) : this._initLatLng(arguments[0], arguments[1]);